warmrestart/warmRestartHelper.cpp

```diff
@@ -293,6 +299,18 @@ bool WarmStartHelper::compareAllFV(const std::vector<FieldValueTuple> &v1,
          * this assumption.
          */
         assert(v1Iter != v1Map.end());
+#endif
+        if (v1Iter == v1Map.end())
+        {
+            /* 
+             * New field added for the  refresh entry
+             * hence return 'no match'
+             */
```

Comment:
@nkelapur please take a look at this. To be unblocked, can we add a specific skip on assert condition for the extra attribute (`weight`) that caused the issue?
I agree that ignoring all the extra attributes can bring us more unwarranted issues. Specific ignore helps us control what exactly is allowed/denied functionaly.

Reply:
@vaibhavhd  I don't completely agree with this change.  
1> this warm-restart helper class is mainly used in fpmsyncd. Not used anywhere else.
2> We are in full control of what is going into app-db, in this case for route entries.
I think this whole point of assert was the design assumption earlier was the the fields before and after warm-restart will same. 
However that design does not apply anymore, since the number of fields is different before and after warm-upgrade. So in this case the general design should be that this api returns "true" such that after warm-reboot reconciliation the new entry will get written to the app-db. This change is implementing it.

If we check the field for a particular field name ex "weight" in this case, then that will make this code non-modular and difficult to maintain. 
At a later point when more new fields are added ex next-hop-group, then another check has to be added to check for that field name.
Also if we plan to use this for something other than route entries, then there will be fields specific to that table we might need to skip.

Please note. We are not ignoring all the new attributes. This api will return true if it finds a non matching attribute, meaning that the entry is different and needs to be set to the app-db at reconcillation. Thus app-db will be updated with the new entry with the extra new attribute and that will trigger OA to take further action.

Reply:
@nkelapur I believe this issue should not be fixed in SONiC. When an new attribute is needed, SAI/SDK should add it in during warm recovery, with SONiC adding it with db_migrator, we have a complete loop.

Reply:
@yxieca Completely agree with this. If db_migrator take care of adding the new attribute, then we don't need to remove this assert.